accounts-db/src/accounts_hash.rs

```diff
@@ -781,47 +781,44 @@ impl<'a> AccountsHasher<'a> {
         // b. lamports
         let _guard = self.active_stats.activate(ActiveStatItem::HashDeDup);
 
+        #[derive(Default)]
+        struct DedupResult {
+            hashes_files: Vec<AccountHashesFile>,
+            hashes_count: usize,
+            lamports_sum: u64,
+        }
+
         let mut zeros = Measure::start("eliminate zeros");
-        let (hashes, hash_total, lamports_total) = (0..max_bin)
+        let DedupResult {
+            hashes_files: hashes,
+            hashes_count: hash_total,
+            lamports_sum: lamports_total,
+        } = (0..max_bin)
             .into_par_iter()
-            .fold(
-                || {
-                    (
-                        /*hashes files*/ Vec::with_capacity(max_bin),
-                        /*hashes count*/ 0_usize,
-                        /*lamports sum*/ 0_u64,
-                    )
-                },
-                |mut accum, bin| {
-                    let (hashes_file, lamports_bin) = self.de_dup_accounts_in_parallel(
-                        sorted_data_by_pubkey,
-                        bin,
-                        max_bin,
-                        stats,
-                    );
-                    accum.2 = accum
-                        .2
-                        .checked_add(lamports_bin)
-                        .expect("summing capitalization cannot overflow");
-                    accum.1 += hashes_file.count();
-                    accum.0.push(hashes_file);
-                    accum
-                },
-            )
+            .fold(DedupResult::default, |mut accum, bin| {
```

Comment:
For some reason, ledger-tool runs OOM on my box. Not sure what went wrong. It was working yesterday. :(

Reply:
Timing measurement from ledger tool:
master: eliminate_zeros_us                  4610293i
this:      eliminate_zeros_us                  4708397i

90 mili-seconds slower, maybe worth for the less usage of memory?

Reply:
Yeah. It would be difficult to get accurate memory measurement from ledger-tool run. However, luckily, a memory-stat data point was reported during dedup. 

The following are the memory-stat reported - used_bytes=**33640493056** vs **34861576192** - seems it uses 1G less memory.

this pr
`[2023-09-29T17:39:35.618258136Z INFO  solana_metrics::metrics] datapoint: memory-stats total=269907464192i swap_total=2044719104i free_percent=5.979692626995669 used_bytes=33640493056i avail_percent=87.53628649852023 buffers_percent=0.24545272728312942 cached_percent=79.40825196280461 swap_free_percent=98.70512561122919
`

master
`[2023-09-29T17:53:22.844732061Z INFO  solana_metrics::metrics] datapoint: memory-stats total=269907464192i swap_total=2044719104i free_percent=8.426771729373367 used_bytes=34861576192i avail_percent=87.08387843353563 buffers_percent=0.19852986044832857 cached_percent=76.63884220588038 swap_free_percent=98.78204884224527
`